frontend/client_sdk/src/CodaSDK.re

```diff
@@ -282,8 +296,54 @@ let signStakeDelegation =
             "delegationPayload": {
               "newDelegate": stakeDelegation.to_,
               "delegator": stakeDelegation.from,
-            }
+            },
           },
         )##signature,
     };
   };
+
+[@bs.send]
+external verifyPaymentSignature: (codaSDK, signed_payment_js) => int =
+  "verifyPaymentSignature";
+
+/**
+  * Verifies a signed payment.
+  *
+  * @param signedPayment - A signed payment transaction
+  * @returns True if the `signed(payment)` is a verifiable payment
+  */
+[@genType]
+let verifyPaymentSignature = (signedPayment: signed(payment)) => {
+  let payload = signedPayment.payload;
+  // Stringify all numeric inputs since they may be passed as
+  // number/bigint in TS/JS
+  let memo = value(~default="", payload.memo);
+  let fee = Js.String.make(payload.fee);
+  let amount = Js.String.make(payload.amount);
+  let nonce = Js.String.make(payload.nonce);
+  let validUntil =
+    Js.String.make(value(~default=defaultValidUntil, payload.validUntil));
+
+  verifyPaymentSignature(
+    codaSDK,
+    {
+      "sender": signedPayment.publicKey,
+      "signature": signedPayment.signature,
+      "payment": {
+        "common": {
+          "fee": fee,
+          "feePayer": payload.from,
+          "nonce": nonce,
+          "validUntil": validUntil,
+          "memo": memo,
+        },
+        "paymentPayload": {
+          "source": payload.from,
+          "receiver": payload.to_,
+          "amount": amount,
+        },
+      },
+    },
+  )
+  ->bool_of_int;
```

Comment:
I'm not sure why calling `verifyPaymentSignature` is returning a number instead of a bool. In `Client_sdk.ml`, the function is specified to return a boolean so I'm not sure if there is any weird casting being done from ocaml -> js.

Reply:
what is the return type for `verifyMessage`? 

if that returns a boolean, stick with that pattern